src/CppParser/Parser.cpp

```diff
@@ -3541,6 +3543,18 @@ Declaration* Parser::WalkDeclaration(const clang::Decl* D,
 
         auto Class = WalkRecordCXX(RD);
 
+        if (WalkRedecls && Class != nullptr)
```

Comment:
Is this check for null, as well as the one below, really necessary now that we handle injected classes?